servers/src/main/java/tachyon/worker/file/FileDataManager.java

```diff
@@ -75,13 +77,20 @@ public synchronized void persistFile(long fileId, List<Long> blockIds) throws IO
     OutputStream outputStream = mUfs.create(dstPath);
     final WritableByteChannel outputChannel = Channels.newChannel(outputStream);
 
+    Map<Long, Long> blockIdToLockId = Maps.newHashMap();
+    // lock all the blocks to prevent any eviction
     for (long blockId : blockIds) {
       long lockId;
       try {
         lockId = mBlockDataManager.lockBlock(Sessions.CHECKPOINT_SESSION_ID, blockId);
+        blockIdToLockId.put(blockId, lockId);
       } catch (BlockDoesNotExistException e) {
         throw new IOException(e);
       }
+    }
```

Comment:
I see. That's very good concern. The general problem is how to lock a set of blocks, i.e. a higher level granularity of locking. Do you have suggestions of better practices?